h/api/models/document.py

```diff
@@ -26,8 +26,12 @@ class Document(Base, mixins.Timestamps):
     #        dependency on the annotator-store is removed, as it clashes with
     #        making the Postgres and Elasticsearch interface of a Document
     #        object behave the same way.
-    document_uris = sa.orm.relationship('DocumentURI', backref='document')
-    meta = sa.orm.relationship('DocumentMeta', backref='document')
+    document_uris = sa.orm.relationship('DocumentURI',
+                                        backref='document',
+                                        order_by='DocumentURI.updated.desc()')
```

Comment:
Added an index to both DocumentMeta/URI `updated` column.

When checking the query with `EXPLAIN ANALYZE ...` I saw that it isn't using an index for filtering the `document_id` column. Turns out Postgres is [not automatically adding indices for foreign keys](https://hyp.is/AVSqPuhUFtPwhMO7Dnnm/www.postgresql.org/docs/9.4/static/ddl-constraints.html). There is a migration for that as well now.